pkg/apis/hive/v1alpha1/dnsendpoint_types.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DNSEndpointSpec defines the desired state of DNSEndpoint
+type DNSEndpointSpec struct {
+	// Endpoints is the list of DNS records to create/update
+	// +optional
```

Comment:
Should this actually be optional?

Reply:
Or is this part of that very specific CRD structure external-dns needs.

Reply:
An empty set of records is valid and useful depending on the sync mode you're using. If you are using sync mode (right now we do upsert), have records you have created previously, and you want them deleted, you'd set the endpoints to empty.